modules/FindBasisUniversal.cmake

```diff
@@ -115,100 +73,162 @@ macro(_basis_setup_source_file source)
+                    HINTS "${BASIS_UNIVERSAL_DIR}/transcoder" "${BASIS_UNIVERSAL_DIR}"
+                    NO_CMAKE_FIND_ROOT_PATH)
+                if(BasisUniversalTranscoder_DIR)
+                    set(BasisUniversal_Transcoder_FOUND TRUE)
+                else()
+                    message(SEND_ERROR
+                        "Could not find transcoder in basis_universal directory.\n"
+                        "Set BASIS_UNIVERSAL_DIR to the root of a directory containing basis_universal source.")
+                endif()
+                set(BasisUniversalTranscoder_SOURCES
+                    ${BasisUniversalTranscoder_DIR}/basisu_transcoder.cpp)
+
+                foreach(_file ${BasisUniversalTranscoder_SOURCES})
+                    _basis_setup_source_file(${_file})
+                endforeach()
+
+                # Disable the find root path here, it overrides the
+                # CMAKE_FIND_ROOT_PATH_MODE_INCLUDE setting potentially set in
+                # toolchains.
+                find_path(BasisUniversalTranscoder_INCLUDE_DIR NAMES basisu_transcoder.h
+                    HINTS "${BASIS_UNIVERSAL_DIR}/transcoder" "${BASIS_UNIVERSAL_DIR}"
+                    NO_CMAKE_FIND_ROOT_PATH)
+                mark_as_advanced(BasisUniversalTranscoder_INCLUDE_DIR)
+
+                add_library(BasisUniversal::Transcoder INTERFACE IMPORTED)
+                set_property(TARGET BasisUniversal::Transcoder APPEND PROPERTY
+                    INTERFACE_INCLUDE_DIRECTORIES ${BasisUniversalTranscoder_INCLUDE_DIR})
+                set_property(TARGET BasisUniversal::Transcoder APPEND PROPERTY
+                    INTERFACE_COMPILE_DEFINITIONS "BASISU_NO_ITERATOR_DEBUG_LEVEL")
+                set_property(TARGET BasisUniversal::Transcoder APPEND PROPERTY
+                    INTERFACE_SOURCES "${BasisUniversalTranscoder_SOURCES}")
             endif()
-            set(BasisUniversalTranscoder_SOURCES
-                ${BasisUniversalTranscoder_DIR}/basisu_transcoder.cpp)
-
-            foreach(_file ${BasisUniversalTranscoder_SOURCES})
-                _basis_setup_source_file(${_file})
-            endforeach()
-
-            add_library(BasisUniversal::Transcoder INTERFACE IMPORTED)
-            set_property(TARGET BasisUniversal::Transcoder APPEND PROPERTY
-                INTERFACE_INCLUDE_DIRECTORIES ${BasisUniversalTranscoder_INCLUDE_DIR})
-            set_property(TARGET BasisUniversal::Transcoder APPEND PROPERTY
-                INTERFACE_COMPILE_DEFINITIONS "BASISU_NO_ITERATOR_DEBUG_LEVEL")
-            set_property(TARGET BasisUniversal::Transcoder APPEND PROPERTY
-                INTERFACE_SOURCES "${BasisUniversalTranscoder_SOURCES}")
         else()
             set(BasisUniversal_Transcoder_FOUND TRUE)
         endif()
     endif()
 endforeach()
 
-include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(BasisUniversal
     REQUIRED_VARS BasisUniversalTranscoder_INCLUDE_DIR BasisUniversalEncoder_INCLUDE_DIR
```

Comment:
This started failing in the MinGW CI build (which doesn't build the encoder AFAIK), moreover this fails in case only either the encoder or transcoder is found (for example, for Habitat, I'm bundling just the transcoder sources instead of the whole thing).

I think keeping just `HANDLE_COMPONENTS` would work, as the component `_FOUND` variables are set properly above (but please verify that it actually does fail properly when nothing is found :)).

Reply:
You mean `HANDLE_COMPONENTS` without the two include dirs?

Reply:
yeah,

```cmake
find_package_handle_standard_args(BasisUniversal
    HANDLE_COMPONENTS)
```